app/controllers/teams_controller.rb

```diff
@@ -53,39 +18,13 @@ def show
   def dissolve
     @team = Team.find(params[:id])
     @team.dissolve(DateTime.current)
-    redirect_back fallback_location: course_teams_path(@course)
-  end
-
-
-  def dissolve_all
-    teams = Team.where(course: @course, end_date: nil)
-    teams.each do |t| t.dissolve(DateTime.current) end
-    redirect_back fallback_location: course_teams_path(@course), notice: "#{plural(teams.count, 'team')} dissolved"
-  end
-
-  def randomize
-    count = 0
-    students_without_active_team.to_a.shuffle.each_slice(params[:random][:size].to_i).each do |t|
-      @team = Team.new(course: @course, start_date: params[:random][:start_date], end_date: params[:random][:end_date])
-      @team.users = t
-
-      if @team.save
-        count += 1
-      end
-    end
-    redirect_back fallback_location: course_teams_path(@course), notice: "#{plural(count, 'random team')} created"
+    redirect_back fallback_location: course_teamsets_path(@course, @teamset),
+                  notice: "Team was successfully dissolved"
   end
 
   private
 
-  def students_without_active_team
-    # TODO: Optimize.
-    @course.students_with_drop_info.where("registrations.dropped_date": nil).reject do |student|
-      student.teams.where(course: @course).any? { |t| t.active? }
-    end
-  end
-
-  def team_params
-    params.require(:team).permit(:course_id, :start_date, :end_date)
+  def find_teamset
+    @teamset = Teamset.find(params[:teamset_id])
```

Comment:
That should be an `Audit.log`able offense, and you'd get bounced back to the nearest enclosing category#index page.  (It isn't, yet, but it should be.)